orderer/consensus/consensus.go

```diff
@@ -26,6 +26,16 @@ type Consenter interface {
 	HandleChain(support ConsenterSupport, metadata *cb.Metadata) (Chain, error)
 }
 
+// JoinBlockValidator performs validation on a join block for the channel participation API.
+type JoinBlockValidator interface {
```

Comment:
If we could push the processing of this validation closer to the client side processing then maybe we wouldn't have to extend this interface.